### PR TITLE
Add RayJob docs and development docs

### PR DESCRIPTION
**Pull request by Jeffwan**
part of #349 

## Why are these changes needed?

- Add RayJob and development docs 
- make index doc cleaner and move to corresponding sections.

Testing

```
docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material
```

Open `localhost:8000`

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,15 +3,13 @@
 [![Build Status](https://github.com/ray-project/kuberay/workflows/Go-build-and-test/badge.svg)](https://github.com/ray-project/kuberay/actions)
 [![Go Report Card](https://goreportcard.com/badge/github.com/ray-project/kuberay)](https://goreportcard.com/report/github.com/ray-project/kuberay)
 
-KubeRay is an open source toolkit to run Ray applications on Kubernetes.
-
-KubeRay provides several tools to improve running and managing Ray's experience on Kubernetes.
+KubeRay is an open source toolkit to run Ray applications on Kubernetes. It provides several tools to improve running and managing Ray's experience on Kubernetes.
```
Comment (DmitriGekhtman) [inline README.md:6]: Suggestion: `Ray's experience` -> `Ray`

File: docs/guidance/rayjob.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+> Note: This is the alpha version of Ray Job Support in KubeRay. There will be ongoing improvements for Ray Job in the future releases.
+
+### Prerequisite
+
+* Ray 1.10 and above.
+* KubeRay v0.3.0 or master
+
+### What is a RayJob?
+
+The RayService is a new custom resource (CR) supported by KubeRay in v0.3.0.
```
Comment (DmitriGekhtman) [inline docs/guidance/rayjob.md:12]: Find and replace
RayService -> RayJob
in this doc.

File: docs/guidance/rayjob.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+A RayService manages 2 things:
+* RayCluster: Manages resources in kubernetes cluster.
+* Ray Serve Deployment Graph: Manages users' serve deployment graph.
```
Comment (DmitriGekhtman) [inline docs/guidance/rayjob.md:16]: We should probably link to the Ray docs to explain what a Ray Serve Deployment Graph is.
cc @simon-mo @brucez-anyscale

File: docs/guidance/rayjob.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
+## Ray Job (alpha)
```
Comment (DmitriGekhtman) [inline docs/guidance/rayjob.md:1]: We should explain what the operational difference is between a Ray Service and a Ray Job.

**Comment by DmitriGekhtman** [comment]
cc @harryge00

**Review by wilsonwang371 — APPROVED** [review]
lgtm

File: docs/guidance/rayjob.md (hunk 2 above)
Comment (Jeffwan) [reply docs/guidance/rayjob.md:12]: Yeah will do

File: docs/guidance/rayjob.md (hunk 3 above)
Comment (Jeffwan) [reply docs/guidance/rayjob.md]: I originally copy some contents and structure from RayService and forget to clean them up. Now it's in good shape

File: docs/guidance/rayjob.md (hunk 4 above)
Comment (Jeffwan) [reply docs/guidance/rayjob.md:1]: Add some details. The cluster can be deleted by RayJob controller once it succeed or failed

**Comment by Jeffwan** [comment]
@DmitriGekhtman @harryge00 Please have another check. it just have RayJob content now

**Comment by harryge00** [comment]
lgtm

**Review by DmitriGekhtman — APPROVED** [review]

**Pull request merged**